tools/ZydisFuzzShared.c

```diff
@@ -175,7 +175,7 @@ void ZydisValidateEnumRanges(const ZydisDecodedInstruction* insn,
             ZYDIS_CHECK_ENUM(op->mem.segment, ZYDIS_REGISTER_MAX_VALUE);
             ZYDIS_CHECK_ENUM(op->mem.base, ZYDIS_REGISTER_MAX_VALUE);
             ZYDIS_CHECK_ENUM(op->mem.index, ZYDIS_REGISTER_MAX_VALUE);
-            ZYDIS_CHECK_ENUM(op->mem.disp.has_displacement, ZYAN_TRUE);
+            ZYDIS_CHECK_ENUM(op->mem.disp.size, ZYDIS_MAX_DISP_IMM_SIZE);
```

Comment:
Not sure if `ZYDIS_CHECK_ENUM()` is appropriate here, as the size has definition holes.

@mappzor Should we as well add checks for `disp.offset`, `imm.offset` and `imm.size`?

Reply:
I would go as strictly as possible:

- displacement size must be one of 0, 8, 32
- immediate size must be one of 0, 8, 16, 32, 64
- offset + (size / 8) <= instruction length

Reply:
@mappzor Because of technical implementation details, we currently have to accept `0, 8, 16, 32, 64` for DISP as well, because I've choosen to use DISP to map the MOFFS operand type:

```
A1 12 34 56 78 12 34 56 78
:  :..DISP
:..OPCODE
```